build_tools/cmake/iree_run_module_test.cmake

```
@@ -0,0 +1,369 @@
+#     "android-arm64-v8a"
+# )
+#
+# iree_benchmark_suite_module_test(
+#   NAME
+#     mobilenet_v1_fp32_correctness_test
+#   BENCHMARK_MODULE_SRC
+#     "TFLite/MobileNetV1-fp32,imagenet"
+#   DRIVER
+#     "local-sync"
+#   RUNNER_ARGS
+#     "--entry_function=main"
+#     "--function_input=1x224x224x3xf32=0"
+#   EXPECTED_OUTPUT
+#     "mobilenet_v1_fp32_expected_output.txt"
+#   UNSUPPORTED_PLATFORMS
+#     "android-arm64-v8a"
+#     "riscv32-Linux"
+# )
+function(iree_benchmark_suite_module_test)
+if(NOT IREE_BUILD_TESTS)
+    return()
+  endif()
+
+  cmake_parse_arguments(
+    _RULE
+    ""
+    "NAME;BENCHMARK_MODULE_SRC;DRIVER;EXPECTED_OUTPUT;TIMEOUT"
+    "RUNNER_ARGS;LABELS;XFAIL;UNSUPPORTED_PLATFORMS"
+    ${ARGN}
+  )
+
+  if(CMAKE_CROSSCOMPILING AND "hostonly" IN_LIST _RULE_LABELS)
+    return()
+  endif()
+
+  iree_get_platform(_PLATFORM)
+  if(_PLATFORM IN_LIST _RULE_UNSUPPORTED_PLATFORMS)
+    return()
+  endif()
+
+  # Benchmark suite needs to be stored at the location of
+  # `IREE_BENCHMARK_SUITE_DIR` or the test target is bypassed.
+  if(_RULE_BENCHMARK_MODULE_SRC AND NOT DEFINED IREE_BENCHMARK_SUITE_DIR)
+    return()
+  endif()
+
+  set(_MODULE_FLAG_DIR "${IREE_BENCHMARK_SUITE_DIR}/${_RULE_BENCHMARK_MODULE_SRC}/")
+  # Find the platform specific module flag file with matching path name.
+  # TODO(#10391): Update this logic with the new benchmark framework.
+  if(_PLATFORM STREQUAL "riscv64-Linux")
+    set(_FLAGFILE_HINT_PATH "${_MODULE_FLAG_DIR}/iree-llvm-cpu*RV64*__full-inference,default-flags/flagfile")
+  elseif(_PLATFORM STREQUAL "riscv32-Linux")
+    set(_FLAGFILE_HINT_PATH "${_MODULE_FLAG_DIR}/iree-llvm-cpu*RV32*__full-inference,default-flags/flagfile")
+  elseif(_PLATFORM STREQUAL "android-arm64-v8a")
+    set(_FLAGFILE_HINT_PATH "${_MODULE_FLAG_DIR}/iree-llvm-cpu*ARM64-v8A*__big-core,full-inference,default-flags/flagfile")
+  else()  # X86_64
+    set(_FLAGFILE_HINT_PATH "${_MODULE_FLAG_DIR}/iree-llvm-cpu*x86_64*__full-inference,default-flags/flagfile")
+  endif()
+  file(GLOB _FLAGFILE_PATH
+      LIST_DIRECTORIES FALSE
+      "${_FLAGFILE_HINT_PATH}"
+    )
+  if(_FLAGFILE_PATH)
+    list(LENGTH _FLAGFILE_PATH _FLAGFILE_NUM)
+    if(_FLAGFILE GREATER 1)
+      message(SEND_ERROR "Found multiple flagfile with '${_FLAGFILE_HINT_PATH}' for ${_RULE_BENCHMARK_MODULE_SRC}")
+    endif()
+    get_filename_component(_FLAG_FILE_DIR "${_FLAGFILE_PATH}" DIRECTORY)
+    file(STRINGS "${_FLAGFILE_PATH}" _FLAGS ENCODING UTF-8)
+    # Parse the flagfile to find the vmfb location.
+    # TODO(#10391): Update this logic with the new benchmark framework.
+    foreach(_FLAG ${_FLAGS})
+      if(_FLAG MATCHES "--module_file=")
+        string(REPLACE "--module_file=" "" _SRC "${_FLAG}")
+        set(_SRC "${_FLAG_FILE_DIR}/${_SRC}")
+      endif()
+    endforeach(_FLAG)
+  else()
+    message(SEND_ERROR "Could not locate flagfile matching '${_FLAGFILE_HINT_PATH}' for ${_RULE_BENCHMARK_MODULE_SRC}")
+  endif(_FLAGFILE_PATH)
```

Incorporating all suggestions from above.

```suggestion
  if(NOT _FLAGFILE_PATH)
    message(SEND_ERROR "Could not locate flagfile matching '${_FLAGFILE_HINT_PATH}' for ${_RULE_BENCHMARK_MODULE_SRC}")
    return()
  endif()
  list(LENGTH _FLAGFILE_PATH _FLAGFILE_NUM)
  if(_FLAGFILE GREATER 1)
    message(SEND_ERROR "Found multiple files matching '${_FLAGFILE_HINT_PATH}' for ${_RULE_BENCHMARK_MODULE_SRC}: ${_FLAGFILE_PATH}")
  endif()
  get_filename_component(_FLAG_FILE_DIR "${_FLAGFILE_PATH}" DIRECTORY)
  file(STRINGS "${_FLAGFILE_PATH}" _FLAGS ENCODING UTF-8)
  # Parse the flagfile to find the vmfb location.
  # TODO(#10391): Update this logic with the new benchmark framework.
  foreach(_FLAG ${_FLAGS})
    if(_FLAG MATCHES "--module_file=")
      string(REPLACE "--module_file=" "" _SRC "${_FLAG}")
      set(_SRC "${_FLAG_FILE_DIR}/${_SRC}")
    endif()
  endforeach(_FLAG)
```

---

Done. Also update the `get_filename_component` usage here.